log4j-core-test/src/test/java/org/apache/logging/log4j/core/filter/HttpThreadContextMapFilterTest.java

```diff
@@ -16,183 +16,98 @@
+            final LoggerContext loggerContext, @Named("List") final ListAppender appender) throws Exception {
 
-    @BeforeAll
-    public static void startServer() throws Exception {
-        try {
-            server = new Server(0);
-            final ServletContextHandler context = new ServletContextHandler();
-            final ServletHolder defaultServ = new ServletHolder("default", TestServlet.class);
-            defaultServ.setInitParameter("resourceBase", System.getProperty("user.dir"));
-            defaultServ.setInitParameter("dirAllowed", "true");
-            context.addServlet(defaultServ, "/");
-            server.setHandler(context);
+        // Create the logger
+        final ExtendedLogger logger = loggerContext.getLogger(HttpThreadContextMapFilterTest.class);
+
+        // Create a response transformer; the only way to dynamically construct WireMock responses.
+        // We need a dynamic response generation, since there we will issue MDC changes and log statements.
+        final ResponseTransformer wireMockResponseTransformer = new ResponseTransformer() {
 
-            // Start Server
-            server.start();
-            port = ((ServerConnector) server.getConnectors()[0]).getLocalPort();
-            try {
-                Files.deleteIfExists(target);
-            } catch (IOException ioe) {
-                // Ignore this.
+            private final AtomicInteger invocationCounter = new AtomicInteger();
+
+            @Override
+            public Response transform(
+                    final Request request,
+                    final Response response,
+                    final FileSource files,
+                    final Parameters parameters) {
+                final int invocationCount = invocationCounter.getAndIncrement();
+                ThreadContext.put("invocationCount", "" + invocationCount);
+                logger.info("transforming request #{}", invocationCount);
+                return response;
             }
-        } catch (Throwable ex) {
-            ex.printStackTrace();
-            throw ex;
-        }
-    }
 
-    @AfterAll
-    public static void stopServer() throws Exception {
-        server.stop();
-    }
+            @Override
+            public String getName() {
+                return "mdc-writer";
+            }
+        };
 
-    @AfterEach
-    public void after() {
-        try {
-            Files.deleteIfExists(target);
-        } catch (IOException ioe) {
-            // Ignore this.
-        }
-        loggerContext.stop();
-        loggerContext = null;
-    }
+        // Create the WireMock server extended using the response transformer.
+        final WireMockServer wireMockServer =
+                new WireMockServer(WireMockConfiguration.wireMockConfig().extensions(wireMockResponseTransformer));
+        wireMockServer.stubFor(get("/").willReturn(ok().withTransformers(wireMockResponseTransformer.getName())));
```

Comment:
WireMock could use an ephemeral port to prevent port collisions. E.g. on my computer port `8080` is already taken.

Reply:
Fixed in 4838ceba9a098b78c51d7fe16bcd4e208fda85c6.